microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -79,6 +91,63 @@ def run_tests(
 
         assert_that(failures, f"Unexpected failures: {failures}").is_empty()
 
+    def run_metrics_tests(
+        self,
+        test_result: TestResult,
+        environment: Environment,
+        hypervisor: str,
+        log_path: Path,
+        skip: Optional[List[str]] = None,
+    ) -> None:
+        self.per_mtr_report_file = log_path.joinpath("perf_metrics.json")
+
+        perf_metrics_tests = self._list_perf_metrics_tests(hypervisor=hypervisor)
+        testcases_result_list: List[CHPerfMetricTestResult] = []
+
+        for testcase in perf_metrics_tests:
+            testcase_result = CHPerfMetricTestResult()
+            testcase_result.name = testcase
+
+            try:
+                result = self.run(
+                    f"tests --hypervisor {hypervisor} --metrics -- -- \
+                        --test-filter {testcase}",
+                    timeout=self.TIME_OUT,
+                    force_run=True,
+                    cwd=self.repo_root,
+                    no_info_log=False,  # print out result of each test
+                    shell=True,
+                )
+                output = result.stdout.replace("\r\n", "\n")
+                output = output.replace("\t", "")
+                if result.exit_code != 0:
+                    raise Exception(f"Testcase failed : {testcase}", output)
+                metrics = self._process_perf_metric_test_result(result.stdout)
+                testcase_result.status = TestStatus.PASSED
+                testcase_result.metrics = metrics
+
+            except Exception as e:
+                self._log.error(f"Testcase failed, tescase name: {testcase}")
+                testcase_result.status = TestStatus.FAILED
+                testcase_result.error = str(e.args[0])
+                testcase_result.trace = str(e.args[1])
+            testcases_result_list.append(testcase_result)
+
+        self._create_perf_metric_report(testcases_result_list)
+        for testcase_result in testcases_result_list:
```

Comment:
It doesn't need a separated for loop to dump results. Send notification in the same loop. So, the results can be sent out as soon as possible.

Reply:
Sure Chi. Will push send notification to _create_perf_metric_report. Thanks

Reply:
Done.

Reply:
Ok Chi. Will make this change